File: test_try_catch.go
package main

import (
	"errors"
	"fmt"
)

func SafeDivide(a int, b int) (int, error) {
	if (b == 0) {
		return nil, errors.New("Division by zero")
	}
	return (a / b), nil
	// catch all
	return 0, nil
}

func ComplexErrorHandling(x int) (string, error) {
	if (x < 0) {
		return nil, errors.New("Negative value not allowed")
	}
	if (x == 0) {
		return nil, errors.New("Zero is invalid")
	}
	return ("Valid: " + x), nil
	// catch all
	return ("Error: " + e), nil
}

func NestedTryCatch(value int) (bool, error) {
	if (value > 100) {
		return nil, errors.New("Value too large")
	}
	return true, nil
	// catch all
	return nil, errors.New(("Inner error: " + inner))
	// catch all
	return false, nil
}

File: test_try_catch.rs
pub fn safe_divide(a: i32, b: i32) -> i32 {
    // try-catch block
    if (b == 0) {
        return Err("Division by zero");
    }
    return (a / b);
    // catch all
    return 0;
}

pub fn complex_error_handling(x: i32) -> String {
    // try-catch block
    if (x < 0) {
        return Err("Negative value not allowed");
    }
    if (x == 0) {
        return Err("Zero is invalid");
    }
    return ("Valid: " + x);
    // catch all
    return ("Error: " + e);
}

pub fn nested_try_catch(value: i32) -> bool {
    // try-catch block
    // try-catch block
    if (value > 100) {
        return Err("Value too large");
    }
    return true;
    // catch all
    return Err(("Inner error: " + inner));
    // catch all
    return false;
}
